eth/stagedsync/stage_mining_create_block.go

```diff
@@ -287,6 +311,16 @@ func SpawnMiningCreateBlockStage(s *StageState, tx kv.RwTx, cfg MiningCreateBloc
 		}
 	}
 
+	if isTrans {
```

Comment:
I'd move this code up, right after `cfg.engine.Prepare`, to short-circuit all the uncles logic.